### PR TITLE
feat(lane_change): add unit test for normal lane change class (RT1-7970)

### PR DESCRIPTION
**Pull request by zulfaqar-azmi-t4**
## Description

Add simple test for normal lane change class.

## Local coverage result before PR

screenshot-20241018-062915Z

## Local coverage result after PR

screenshot-20241018-062602Z

## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

**Review by maxime-clem — APPROVED** [review]

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    rclcpp::shutdown();
+  }
+
+  LCParamPtr lc_param_ptr_;
+  std::shared_ptr<NormalLaneChange> normal_lane_change_;
+  std::shared_ptr<PlannerData> planner_data_ = std::make_shared<PlannerData>();
+  LaneChangeModuleType lc_type_{LaneChangeModuleType::NORMAL};
+  Direction lc_direction_{Direction::RIGHT};
+  std::string name = "test_lane_change_scene";
+  std::string test_utils_dir{"autoware_test_utils"};
```
Comment (mkquda) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp:165]:
```suggestion
  std::string name_{"test_lane_change_scene"};
  std::string test_utils_dir_{"autoware_test_utils"};
```

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    normal_lane_change_->setData(planner_data_);
+    set_previous_approved_path();
+  }
+
+  [[nodiscard]] const CommonDataPtr & get_common_data_ptr() const
+  {
+    return normal_lane_change_->common_data_ptr_;
+  }
+
+  [[nodiscard]] rclcpp::NodeOptions init_node_options() const
```
Comment (mkquda) [inline planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp:81]: `get_node_options` seems more appropriate here

**Review by mkquda — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp (hunk 2 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp]: changed in https://github.com/autowarefoundation/autoware.universe/pull/9090/commits/0e362e625dead16c68378d8254b438b9e79c2799

File: planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp (hunk 1 above)
Comment (zulfaqar-azmi-t4) [reply planning/behavior_path_planner/autoware_behavior_path_lane_change_module/test/test_lane_change_scene.cpp]: changed in https://github.com/autowarefoundation/autoware.universe/pull/9090/commits/0e362e625dead16c68378d8254b438b9e79c2799

**Pull request merged**